docs/reference/compute-engine/ray.md

```diff
@@ -0,0 +1,393 @@
+- Applies field mappings and filters
+
+### RayJoinNode
+
+Performs distributed joins:
+- **Broadcast Join**: For small datasets (<100MB)
+- **Distributed Join**: For large datasets with time-based windowing
+- **Automatic Strategy Selection**: Based on dataset size and cluster resources
+
+### RayFilterNode
+
+Applies filters and time-based constraints:
+- TTL-based filtering
+- Timestamp range filtering
+- Custom predicate filtering
+
+### RayAggregationNode
+
+Handles feature aggregations:
+- Windowed aggregations
+- Grouped aggregations
+- Custom aggregation functions
+
+### RayTransformationNode
+
+Applies feature transformations:
+- Row-level transformations
+- Column-level transformations
+- Custom transformation functions
+
+### RayWriteNode
+
+Writes results to various targets:
+- Online stores
+- Offline stores
+- Temporary storage
+
+## Join Strategies
+
+The Ray compute engine automatically selects optimal join strategies:
+
+### Broadcast Join
+
+Used for small feature datasets:
+```python
+# Automatically selected when feature data < 100MB
+# Features are cached in Ray's object store
+# Entities are distributed across cluster
+# Each worker gets a copy of feature data
+```
+
+### Distributed Windowed Join
+
+Used for large feature datasets:
+```python
+# Automatically selected when feature data > 100MB
+# Data is partitioned by time windows
+# Point-in-time joins within each window
+# Results are combined across windows
+```
+
+### Strategy Selection Logic
+
+```python
+def select_join_strategy(feature_size_mb, threshold_mb):
+    if feature_size_mb < threshold_mb:
+        return "broadcast"
+    else:
+        return "distributed_windowed"
+```
+
+## Performance Optimization
+
+### Automatic Optimization
+
+The Ray compute engine includes several automatic optimizations:
+
+1. **Partition Optimization**: Automatically determines optimal partition sizes
+2. **Join Strategy Selection**: Chooses between broadcast and distributed joins
+3. **Resource Allocation**: Scales workers based on available resources
+4. **Memory Management**: Handles out-of-core processing for large datasets
+
+### Manual Tuning
+
+For specific workloads, you can fine-tune performance:
+
+```yaml
+batch_engine:
+    type: ray.engine
+    # Fine-tuning for high-throughput scenarios
+    broadcast_join_threshold_mb: 200      # Larger broadcast threshold
+    max_parallelism_multiplier: 1        # Conservative parallelism
+    target_partition_size_mb: 512        # Larger partitions
+    window_size_for_joins: "2H"          # Larger time windows
+```
+
+### Monitoring and Metrics
+
+Monitor Ray compute engine performance:
+
+```python
+import ray
+
+# Check cluster resources
+resources = ray.cluster_resources()
+print(f"Available CPUs: {resources.get('CPU', 0)}")
+print(f"Available memory: {resources.get('memory', 0) / 1e9:.2f} GB")
+
+# Monitor job progress
+job = store.get_historical_features(...)
+# Ray compute engine provides built-in progress tracking
```

Comment:
I think we missed how to track the progress here in python code or from UI ?


or is it just an information ?

Reply:
It's Ray's dataset and dashboard progress bars, check https://docs.ray.io/en/latest/data/monitoring-your-workload.html#ray-data-progress-bars